### ISSUE TITLE
Wrong result when adding two tensors with indexing and type conversion

### ISSUE THREAD

**Issue by Dobatymo** [pytorch/pytorch#29272]
## 🐛 Bug

Well, the code sample best shows what's going on. I am trying to add one int and one float tensor. And the result is wrong and even yield different results every time.

## To Reproduce

```python
import torch

t1 = torch.tensor([[1,2],[3,4]])
t2 = torch.tensor([0.5, 0.5])
print(t1+t2) # tensor([[1.5, 2.5], [3.5, 4.5]]) correct
print(t1[:,0]) # tensor([1, 3]) correct
print(t1[:,0] + t2) # tensor([1.5000, 0.5000]) whaaaat?
```

Executing this code yields a wrong result for the last addition. The second element even changes if the snippet is executed multiple times. So far I got `0.5`, `4.0` and `nan`.

## Expected behavior

The result of the last line should be `tensor([1.5000, 3.5000])`

## Environment

```
PyTorch version: 1.3.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Microsoft Windows 10 Enterprise
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: GeForce 940MX
Nvidia driver version: 430.86
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] Could not collect
[conda] Could not collect
```
installed using `py -m pip install torch===1.3.0 torchvision===0.4.1 -f https://download.pytorch.org/whl/torch_stable.html`

**Comment by yf225** [comment]
@nairbv Would you like to take a look?

**Comment by nairbv** [comment]
@Dobatymo yes this is a bug in mixed-type operations on non-contiguous tensors. The bug is fixed on master and in 1.3.1.
